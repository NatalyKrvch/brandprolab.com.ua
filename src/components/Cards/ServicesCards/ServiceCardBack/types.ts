import { ServiceCardFrontVariant } from '../ServiceCardFront/types';

export interface ServiceCardBackProps {
  header: string;
  list: string[];
  variant?: ServiceCardFrontVariant;
  backgroundImageUrl?: string;
}
