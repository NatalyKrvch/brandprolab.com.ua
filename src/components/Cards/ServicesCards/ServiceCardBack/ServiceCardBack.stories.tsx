import type { Meta, StoryObj } from '@storybook/react';

import ServiceCardBack from '../ServiceCardBack/ServiceCardBack';

const meta: Meta<typeof ServiceCardBack> = {
  title: 'Components/ServiceCardBack',
  component: ServiceCardBack,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    header: {
      control: 'text',
      description: 'The title displayed at the top of the card.',
      table: {
        type: { summary: 'string' },
      },
    },
    list: {
      control: { type: 'object' },
      description: 'A list of items displayed in the card.',
      table: {
        type: { summary: 'string[]' },
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof ServiceCardBack>;

export const Default: Story = {
  args: {
    header: 'Основні переваги',
    list: [
      'Глибокий аналіз',
      'Професійний підхід',
      'Перевірена інформація',
      'Експертні висновки',
    ],
  },
};
