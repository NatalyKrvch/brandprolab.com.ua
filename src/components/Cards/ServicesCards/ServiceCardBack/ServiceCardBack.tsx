import { SimpleCard } from '@/components/Cards/SimpleCard';
import {
  SERVICE_CARD_BACK_HEADER_TEST_ID,
  SERVICE_CARD_BACK_LIST_ITEM_TEST_ID,
  SERVICE_CARD_BACK_LIST_TEST_ID,
} from '@/lib/testIDs';
import { BACK_CARD_BG_IMAGE_OPACITY } from '@/styles/constants';

import { ServiceCardFrontVariant } from '../ServiceCardFront';
import { serviceCardBackStyleMap } from './styleMaps';
import type { ServiceCardBackProps } from './types';

const ServiceCardBack = ({
  header,
  list,
  backgroundImageUrl,
  variant = ServiceCardFrontVariant.DEFAULT,
}: ServiceCardBackProps) => {
  const bgImageURL =
    variant === ServiceCardFrontVariant.GRADIENT
      ? backgroundImageUrl
      : undefined;
  const styles = serviceCardBackStyleMap[variant];

  return (
    <SimpleCard
      decorativeBgImageUrl={bgImageURL}
      decorativeBgImageOpacity={BACK_CARD_BG_IMAGE_OPACITY}
      className={styles.wrapper}
    >
      <h3
        className={styles.header}
        data-testid={SERVICE_CARD_BACK_HEADER_TEST_ID}
      >
        {header}
      </h3>

      <ul className={styles.list} data-testid={SERVICE_CARD_BACK_LIST_TEST_ID}>
        {list.map(item => (
          <li key={item} data-testid={SERVICE_CARD_BACK_LIST_ITEM_TEST_ID}>
            {item}
          </li>
        ))}
      </ul>
    </SimpleCard>
  );
};

export default ServiceCardBack;
