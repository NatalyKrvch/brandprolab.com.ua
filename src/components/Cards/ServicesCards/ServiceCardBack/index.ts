export { default as ServiceCardBack } from './ServiceCardBack';
