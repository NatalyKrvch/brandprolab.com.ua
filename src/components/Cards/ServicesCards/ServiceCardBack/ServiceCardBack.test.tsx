import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import {
  SERVICE_CARD_BACK_HEADER_TEST_ID,
  SERVICE_CARD_BACK_LIST_ITEM_TEST_ID,
  SERVICE_CARD_BACK_LIST_TEST_ID,
  SIMPLE_CARD_TEST_ID,
} from '@/lib/testIDs';

import ServiceCardBack from './ServiceCardBack';

describe('ServiceCardBack component', () => {
  const defaultProps = {
    header: 'Test Header',
    list: ['Item 1', 'Item 2', 'Item 3'],
  };

  it('renders the ServiceCardBack component', () => {
    const { asFragment } = render(<ServiceCardBack {...defaultProps} />);
    expect(screen.getByTestId(SIMPLE_CARD_TEST_ID)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the header correctly', () => {
    const { asFragment } = render(<ServiceCardBack {...defaultProps} />);
    expect(
      screen.getByTestId(SERVICE_CARD_BACK_HEADER_TEST_ID),
    ).toHaveTextContent('Test Header');
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the list correctly', () => {
    const { asFragment } = render(<ServiceCardBack {...defaultProps} />);
    expect(
      screen.getByTestId(SERVICE_CARD_BACK_LIST_TEST_ID),
    ).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct number of list items', () => {
    const { asFragment } = render(<ServiceCardBack {...defaultProps} />);
    const listItems = screen.getAllByTestId(
      SERVICE_CARD_BACK_LIST_ITEM_TEST_ID,
    );

    expect(listItems).toHaveLength(defaultProps.list.length);
    listItems.forEach((item, index) => {
      expect(item).toHaveTextContent(defaultProps.list[index]);
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
