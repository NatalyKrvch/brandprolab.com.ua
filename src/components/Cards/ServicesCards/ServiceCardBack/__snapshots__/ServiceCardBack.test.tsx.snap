// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCardBack component renders the ServiceCardBack component 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center rounded-24 tablet:rounded-32 desktop:rounded-40 flex h-full flex-col gap-4 bg-blue-dark p-6 tablet:p-8 mid_tablet:border-none desktop:gap-6 desktop:border-none desktop:p-10"
    data-testid="simple-card-test-id"
  >
    <h3
      class="mid_tablet:max-w-600 desktop:max-w-600 text-lg font-bold leading-5 text-teal-medium min_mobile:text-base tablet:text-2xl tablet:leading-6 mid_tablet:text-xl desktop:text-2xl desktop:leading-6"
      data-testid="service-card-back-header-test-id"
    >
      Test Header
    </h3>
    <ul
      class="mid_tablet:max-w-540 desktop:max-w-544 list-disc space-y-1.5 pl-5 text-sm font-extralight leading-18 tracking-normal text-white min_mobile:text-xs tablet:text-base tablet:leading-6 mid_tablet:text-sm mid_tablet:leading-4 desktop:text-lg desktop:leading-5"
      data-testid="service-card-back-list-test-id"
    >
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 1
      </li>
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 2
      </li>
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`ServiceCardBack component renders the correct number of list items 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center rounded-24 tablet:rounded-32 desktop:rounded-40 flex h-full flex-col gap-4 bg-blue-dark p-6 tablet:p-8 mid_tablet:border-none desktop:gap-6 desktop:border-none desktop:p-10"
    data-testid="simple-card-test-id"
  >
    <h3
      class="mid_tablet:max-w-600 desktop:max-w-600 text-lg font-bold leading-5 text-teal-medium min_mobile:text-base tablet:text-2xl tablet:leading-6 mid_tablet:text-xl desktop:text-2xl desktop:leading-6"
      data-testid="service-card-back-header-test-id"
    >
      Test Header
    </h3>
    <ul
      class="mid_tablet:max-w-540 desktop:max-w-544 list-disc space-y-1.5 pl-5 text-sm font-extralight leading-18 tracking-normal text-white min_mobile:text-xs tablet:text-base tablet:leading-6 mid_tablet:text-sm mid_tablet:leading-4 desktop:text-lg desktop:leading-5"
      data-testid="service-card-back-list-test-id"
    >
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 1
      </li>
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 2
      </li>
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`ServiceCardBack component renders the header correctly 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center rounded-24 tablet:rounded-32 desktop:rounded-40 flex h-full flex-col gap-4 bg-blue-dark p-6 tablet:p-8 mid_tablet:border-none desktop:gap-6 desktop:border-none desktop:p-10"
    data-testid="simple-card-test-id"
  >
    <h3
      class="mid_tablet:max-w-600 desktop:max-w-600 text-lg font-bold leading-5 text-teal-medium min_mobile:text-base tablet:text-2xl tablet:leading-6 mid_tablet:text-xl desktop:text-2xl desktop:leading-6"
      data-testid="service-card-back-header-test-id"
    >
      Test Header
    </h3>
    <ul
      class="mid_tablet:max-w-540 desktop:max-w-544 list-disc space-y-1.5 pl-5 text-sm font-extralight leading-18 tracking-normal text-white min_mobile:text-xs tablet:text-base tablet:leading-6 mid_tablet:text-sm mid_tablet:leading-4 desktop:text-lg desktop:leading-5"
      data-testid="service-card-back-list-test-id"
    >
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 1
      </li>
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 2
      </li>
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`ServiceCardBack component renders the list correctly 1`] = `
<DocumentFragment>
  <div
    class="relative w-full bg-cover bg-center rounded-24 tablet:rounded-32 desktop:rounded-40 flex h-full flex-col gap-4 bg-blue-dark p-6 tablet:p-8 mid_tablet:border-none desktop:gap-6 desktop:border-none desktop:p-10"
    data-testid="simple-card-test-id"
  >
    <h3
      class="mid_tablet:max-w-600 desktop:max-w-600 text-lg font-bold leading-5 text-teal-medium min_mobile:text-base tablet:text-2xl tablet:leading-6 mid_tablet:text-xl desktop:text-2xl desktop:leading-6"
      data-testid="service-card-back-header-test-id"
    >
      Test Header
    </h3>
    <ul
      class="mid_tablet:max-w-540 desktop:max-w-544 list-disc space-y-1.5 pl-5 text-sm font-extralight leading-18 tracking-normal text-white min_mobile:text-xs tablet:text-base tablet:leading-6 mid_tablet:text-sm mid_tablet:leading-4 desktop:text-lg desktop:leading-5"
      data-testid="service-card-back-list-test-id"
    >
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 1
      </li>
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 2
      </li>
      <li
        data-testid="service-card-back-list-item-test-id"
      >
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
